### PR TITLE
fix(rbac): adjust access on TeamRole & TeamRoleBindings for org members & cluster admins

### PR DESCRIPTION
**Pull request by IvoGoman**
## Description

Organization members should be able to view TeamRoles and TeamRoleBindings. 

## What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [x] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents

- Related Issue # (issue)
- Closes # (issue)
- Fixes # (issue)

> Remove if not applicable

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help
- [ ] Separate ticket for tests # (issue/pr)

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

## Added to documentation?

- [ ] 📜 README.md
- [ ] 🤝 Documentation pages updated
- [ ] 🙅 no documentation needed
- [ ] (if applicable) generated OpenAPI docs for CRD changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by uwe-mayer — APPROVED** [review]

**Review by uwe-mayer — CHANGES_REQUESTED** [review]
We still need to update org-admin permissions to include TeamRoleBindings..
Blocking this for a moment

**Comment by IvoGoman** [comment]
> We still need to update org-admin permissions to include TeamRoleBindings.. Blocking this for a moment

Since Cluster Admins should be able to manage access on the remote cluster, the access to the TeamRole was moved from the Org Admin to the Cluster Admin.
By default the Org Admin includes all access from the other profiles.

File: pkg/rbac/role.go [hunk 1]
```diff
@@ -109,7 +109,7 @@ func OrganizationMemberPolicyRules() []rbacv1.PolicyRule {
 		{
 			Verbs:     []string{"get", "list", "watch"},
 			APIGroups: []string{greenhouseapisv1alpha1.GroupVersion.Group},
-			Resources: []string{"clusters", "plugins", "pluginpresets", "teams", "teammemberships"},
+			Resources: []string{"clusters", "plugins", "pluginpresets", "teams", "teammemberships", "teamroles", "teamrolebindings"},
```
Comment (uwe-mayer) [inline pkg/rbac/role.go:112]:
```suggestion
			Resources: []string{"*"},
```

File: pkg/rbac/role.go [hunk 2]
```diff
@@ -72,6 +66,12 @@ func OrganizationClusterAdminPolicyRules() []rbacv1.PolicyRule {
 			APIGroups: []string{corev1.GroupName},
 			Resources: []string{"secrets"},
 		},
+		// Grant permission for TeamRoles
+		{
+			Verbs:     []string{"get", "list", "watch", "update", "patch", "delete", "create"},
+			APIGroups: []string{greenhouseapisv1alpha1.GroupVersion.Group},
+			Resources: []string{"teamroles"},
```
Comment (uwe-mayer) [inline pkg/rbac/role.go:73]: Should org-admins not be able to manipulate all Greenhouse resources? So we could lose lines 17-21 of this file?
```suggestion
			Resources: []string{"*"},
```

Comment (uwe-mayer) [reply pkg/rbac/role.go:73]: Ah no. We have to distinguish `organizations` only in orgs namespace...
But still we can consolidate with lines 17-21 and add clusters + plugins
And we should add the List/watch/get of PluginDefinitions to the org-admin as well, right?

File: pkg/rbac/role.go (hunk 1 above)
Comment (uwe-mayer) [reply pkg/rbac/role.go:112]: Same here for the `organization`, disregard.

File: pkg/rbac/role.go (hunk 2 above)
Comment (uwe-mayer) [reply pkg/rbac/role.go:73]:
```suggestion
			Resources: []string{"clusters", "plugins", "pluginpresets", "teams", "teammemberships", "teamroles", "teamrolebindings"},
```

**Review by uwe-mayer — APPROVED** [review]

**Pull request merged**